doc_source/what-is-cloudformation-cli.md

```diff
@@ -1,5 +1,139 @@
+| --- | --- | --- | --- | 
+|  Go  |  General availability  |  [cloudformation\-cli\-go\-plugin](https://github.com/aws-cloudformation/cloudformation-cli-go-plugin/)  |  `cloudformation-cli-go-plugin`  | 
+|  Java  |  General availability  |  [cloudformation\-cli\-java\-plugin](https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/)  |  `cloudformation-cli-java-plugin`  | 
+|  Python  |  General availability  |  [cloudformation\-cli\-python\-plugin](https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/)  |  `cloudformation-cli-python-plugin`  | 
+|  TypeScript  |  General availability  |  [cloudformation\-cli\-typescript\-plugin](https://github.com/aws-cloudformation/cloudformation-cli-typescript-plugin/)  |  `cloudformation-cli-typescript-plugin`  | 
+
+### Upgrading to CFN\-CLI 2\.0<a name="resource-type-setup-upgrade"></a>
+
+If you have developed resource types using the CFN\-CLI 1\.0, we recommend you update to CFN\-CLI 2\.0 and rebuild those types\. Upgrading involves updating the CloudFormation CLI, in addition to any language plugins you are using, but doesn't require any changes to your resource type solution itself\.
+
+Enhancements in CFN\-CLI 2\.0 include:
++ Increased resource payload limit, from 8 KB to 6 MB\.
++ Increased resource stabilization time, from 12 hours to 36 hours, or 48 hours if you are using a stack role to consume the resource\.
++ Improved resource stability, with improved retry strategy and fail\-fast\.
+
+**To upgrade CFN\-CLI 2\.0 and the CloudFormation Provider Development Toolkit Go Plugin**
+
+1. Upgrade the Go Plugin using the following command:
+
+   ```
+   pip3 install --upgrade cloudformation-cli-go-plugin
+   ```
+
+1. Update the Go plugin in the `go.mod` file\.
+
+   ```
+   go get -u github.com/aws-cloudformation/cloudformation-cli-go-plugin 
+   ```
+
+1. To update a resource type to use the CFN\-CLI 2\.0, build and register a new version of the resource using the following command:
+
+   ```
+   make
+   cfn submit --set-default
+   ```
+
+**To upgrade CFN\-CLI 2\.0 and the CloudFormation Provider Development Toolkit Java Plugin**
+
+1. Upgrade the Java Plugin using the following command:
+
+   ```
+   pip3 install --upgrade cloudformation-cli-java-plugin
+   ```
+
+1. Update Java plugin in maven pom\.xml to 2\.0\.0\.
+
+   ```
+   <dependency>
+     <groupId>software.amazon.cloudformation</groupId>
+     <artifactId>aws-cloudformation-rpdk-java-plugin</artifactId>
+     <version>2.0.0</version>
+   </dependency>
+   ```
+
+1. To update a resource type to use the CFN\-CLI 2\.0, build and register a new version of the resource using the following command:
+
+   ```
+   mvn package
+   cfn submit --set-default
+   ```
+
+**To upgrade CFN\-CLI 2\.0 and the CloudFormation Provider Development Toolkit Python Plugin**
+
+1. Upgrade the Python Plugin using the following command:
+
+   ```
+   pip3 install cloudformation-cli-python-plugin
+   ```
```

Comment:
this is the command for installation. Upgrade command is:
`pip3 install --upgrade cloudformation-cli-python-plugin `